### PR TITLE
Add basic documentation for hooks

### LINKED ISSUE #684: Create documentation for hooks
Hooks should be documented on readthedocs.io as a core component. Likely needed to be added to the getting started guide, and have its own section with detailed use cases.

### LINKED ISSUE #714: Add tutorial to `how to use ssh_keys` in hooks
One can use ssh_keys for post provisioning tasks but its not intuitive enough about how to use them. There should be a tutorial for the same. 
Adding details:
The purpose of the tutorial should be how we create ssh_keys/keypairs and how to reference them inside a ansible hook . 
Example workspace:
https://github.com/samvarankashyap/linchpin_hooks_openstack_ws

### LINKED ISSUE #883: Doc: Action managers documentation
Action managers have different action syntax and handle context differently. The details are covered partially inline but should be available in the general documentation. For example [Ansible][https://github.com/CentOS-PaaS-SIG/linchpin/blob/develop/linchpin/hooks/action_managers/ansible_action_manager.py] action syntax is:
```yaml
- playbook: playbook_file
  vars: vars_file
  extra_vars: '{}'
```
While for shell, the action is simply strings.

### PR DESCRIPTION
**Pull request by samvarankashyap**
Summary:

* Basic documentation for hooks
* Command line options for hooks
* Hooks examples

fixes #684
fixes #714
fixes #883

### CODE REVIEW

File: docs/source/custom_action_managers.rst [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+          print("This is the custom hook that runs custom logic")
+
+          for action in self.action_data["actions"]:
+              path = self.action_data["path"]
+              script = action.get("script")
+              print(script)
+              print(path)
+
+In order to write a custom_action_manager one has to implement ActionManager class overriding the following functions:
```
Comment (14rcole) [inline docs/source/custom_action_managers.rst]: is this a duplicate line?

Comment (samvarankashyap) [reply docs/source/custom_action_managers.rst]: Yup ! Will change it.

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**